src/main.lua

```diff
@@ -1701,41 +1714,56 @@ addEventingHandler("distribute", utils.hasMatchingTag("Action", "Tick"), functio
 	local tickedRewardDistributions = {}
 	local totalTickedRewardsDistributed = 0
 
+	print("Ticking from " .. lastTickedEpochIndex .. " to " .. targetCurrentEpochIndex)
+
 	--- tick to the newest epoch, and stub out any epochs that are not yet created
 	for i = lastTickedEpochIndex + 1, targetCurrentEpochIndex do
+		print("Ticking epoch " .. i)
 		local _, _, epochDistributionTimestamp = epochs.getEpochTimestampsForIndex(i)
 		-- use the minimum of the msg timestamp or the epoch distribution timestamp, this ensures an epoch gets created for the genesis block
 		-- and that we don't try and distribute before an epoch is created
 		local tickTimestamp = math.min(msg.Timestamp, epochDistributionTimestamp)
 		-- TODO: if we need to "recover" epochs, we can't rely on just the current message hashchain and block height,
 		-- we should set the prescribed observers and names to empty arrays and distribute rewards accordingly
 		local tickResult = tick.tickEpoch(tickTimestamp, blockHeight, hashchain, msgId)
-		if tickTimestamp == epochDistributionTimestamp then
-			-- if we are distributing rewards, we should update the last ticked epoch index to the current epoch index
-			LastTickedEpochIndex = i
-			table.insert(tickedEpochIndexes, i)
-		end
```

Comment:
this never made sense - if we've ticked the epoch at all, we should increment the `LastTickedEpochIndex`. We can also consider replacing it with two more granular global variables `LastCreatedEpochIndex` and `LastDistributedEpochIndex`

Reply:
we could also add `LastDemandFactorUpdateTimestamp` to track when the demand factor was last updated.

Reply:
merged into his branch on https://github.com/ar-io/ar-io-network-process/pull/317/commits/49661f138b4e8a7d8a75dfe4cbceb76526853f7c